### DISCUSSION TITLE
How SWR works behind the scene

### DISCUSSION THREAD

**Discussion by doptster** [vercel/swr#648] [General] [answered]
Hello,

I understood the basic usages of SWR on the web application.
Before using the library, I would like to know 
1. Where the cache will be stored? Would it impact on the long-term running on the browser, like memory overload etc.
2. Given that my query could be fetching quite a amount of data, should I worry about the memory issue?

**Comment by pke** [comment]
I think the cache lives entirely in memory and additional on disk if the fetch function cached some response data.

When you fetch a lot of data you could think about pagination, since you rarely need all that data ready at any time. Previously fetched data will be discarded by the JS garbage collector if its no longer referenced (by you or swr).

**Comment by sergiodxa** [comment, marked as answer]
Cache is stored solely in memory, there is no size limit (I have a PR open to add a limit to the amount of cached keys).

In my experience, there is no need to worry about the size, most apps don’t fetch enough data to have a memory issue, if using something like Redux or storing in useState you don’t worry then you can probably don’t worry here.

**Reply by doptster** [reply]
Thanks a lot. This answered my doubts in head.
